addons/JobChange/jobchange.lua

```diff
@@ -0,0 +1,145 @@
+
+-- 1.0.1 first release
+-- 1.0.2 added 'reset' command to simply reset to existing job.  Changes sub job to a random starting job and back.
+-- 1.0.3 Code clean-up
+
+require('tables')
+packets = require('packets')
+res = require ('resources')
+
+function jobchange(job,main_sub)
+    windower.add_to_chat(4,"JobChange: Changing "..main_sub.." job to:"..res.jobs[job].ens)
+    if job and main_sub then 
+        local packet = packets.new('outgoing', 0x100, {
+            [main_sub:sub(1,1):upper()..main_sub:sub(2)..' Job'] = job,
+        })
+        packets.inject(packet)
+        coroutine.sleep(0.5)
+    end    
+end
+
+windower.register_event('addon command', function(command, ...)
+    local self = windower.ffxi.get_player()
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
+    end
+    local main_sub = ''
+    if command:lower() == 'main' then
+        main_sub = 'main'
+    elseif command:lower() == 'sub' then
+        main_sub = 'sub'
+    elseif command:lower() == 'reset' then
+        windower.add_to_chat(4,"JobChange: Resetting Job")
+        main_sub = 'sub'
+        job = windower.ffxi.get_player().sub_job:lower()
+    else
+        windower.add_to_chat(4,"JobChange Syntax: //jc main|sub JOB  -- Chnages main or sub to target JOB")
+        windower.add_to_chat(4,"JobChange Syntax: //jc reset -- Resets Current Job")
+        return
+    end
+    local conflict = find_conflict(job,self)
+    local jobid = find_job(job,self)
+    if jobid then 
+        local npc = find_job_change_npc()
+        if npc then
+            if not conflict then 
+                jobchange(jobid,main_sub)
+            else
+                local temp_job = find_temp_job(self)            
+                windower.add_to_chat(4,"JobChange: Conflict with "..conflict)
+                if main_sub == conflict then 
+                    jobchange(temp_job,main_sub)
+                    jobchange(jobid,main_sub)
+                else
+                    jobchange(temp_job,conflict)
+                    jobchange(jobid,main_sub)
+                end
+            end
+        else
+            windower.add_to_chat(4,"JobChange: Not close enough to a Moogle!")
+        end        
+    else
+        windower.add_to_chat(4,"JobChange: Could not change "..command.." to "..job:upper().." ---Mistype|NotUnlocked")
+    end
+    
+end)
+
+function find_conflict(job,self)
+    if self.main_job == job:upper() then
+        return "main"
+    end
+    if self.sub_job == job:upper() then
+        return "sub"
+    end
+end
+
+function find_temp_job(self)
+    local starting_jobs = S { 'WAR', 'MNK', 'WHM', 'BLM', 'RDM', 'THF' } 
+    for i in pairs(starting_jobs) do
+        if not find_conflict(i,self) then 
+            for index,value in pairs(res.jobs) do
+                if value.ens == i then
+                    return index
+                end
+            end
+        end
+    end
+end
+
+function find_job(job,self)
```

Comment:
Again, this function can be shortened quite a bit. First, note what the `jobLevel` variable does. If it's 0, then the loop goes through all jobs, but the condition will always be false. I.e. it's not reliant on the loop variable. In that case you can remove it from the loop entirely:
```lua
function find_job(job, self)
    if self.jobs[job:upper()] == 0 then
        return
    end

    -- loop here
end
```

And like I said before, you could save yourself the `:upper()` if you did it on `args[1]` earlier, since the value is already uppercase from any other source.